driver/driver.go

```diff
@@ -94,7 +95,7 @@ type FlagSet interface {
 
 	// StringList is similar to String but allows multiple values for a
 	// single flag
-	StringList(name string, def string, usage string) *[]*string
+	StringList(name string, def string, usage string) *stringlist.StringList
```

Comment:
This is part of pprof's public interface (one passes in `plugin.Options` to `driver.PProf`; we use this internally to, among other things, change the profile fetcher / symbolizer for internal pprof). 

It would be nice if this interface did not have to change. Since we use this interface internally, it will definitely make this PR simpler to land if there aren't changes to pprof's public interface.

Reply:
Also, this public interface shouldn't use an internal type (`github.com/google/pprof/internal/stringlist`) in its signature.

Reply:
I think the interface needs to change.  Currently there is no difference between a StringList flag and a String flag.  A StringList flag can only hold a single string for now.  To support a list of strings, we need to provide a type that adds strings to a list.  The new type implements the [flag.Value interface](https://golang.org/src/flag/flag.go?s=32508:32557#L291) to achieve this.

> Also, this public interface shouldn't use an internal type (github.com/google/pprof/internal/stringlist) in its signature.

I have moved the new type to a public directory.

Reply:
I think I'd need to see a design document considering this more carefully proposed change before I'd be comfortable changing the public interface. I'm not currently persuaded that the interface change is the best or only way forward; I'd really need to be convinced of that before I'd feel comfortable with a change to the interface. 

As additional context, changing the interface is a really big deal. First, we don't currently version pprof. Now that golang does support versioning a library, if we change the public interface, this becomes something we ought to look into. IMO, it'd be necessary to at least fully consider versioning pprof  using go modules prior to making any breaking changes to the public interface. Next, changes to the public interface can make it really hard to sync our internal copy of pprof, since it becomes possible that code used for our internal version of pprof will break. If it is concluded that changes to the public interface are necessary, I would prefer to first see changes made to our internal copy of pprof to know that nothing internal will break (a PR would still need to be sent to change code, and any changes that need to be made to our dependencies on this library would need to be done when importing pprof from GitHub).

Reply:
I have updated the PR with a different implementation (suggested by lmilewski) that does not change the public interface.  Please take another look.